### PR TITLE
Fixup files where we pin VCPUs for certain domains

### PR DESCRIPTION
**Pull request by otyshchenko1**
Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Suggested-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>

### CODE REVIEW

**Review by lorc — COMMENTED** [review]
I prefer not to use constant "4-7" in config file. Because if there will be need to change pinings, you will be forced to changes those values in two places: in script and in receipt.

Better to put in script some "constant" like DOM0_PIN_CPUS. And introduce two appends:

```
do_install_append_salvator-x-m3-xt() {
￼ +    # As M3 has only 2 A57 cores we have to modify files where we pin VCPUs for certain domains
￼ +
￼ +    # Fixup a number of PCPUs the VCPUs of Dom0 must run on
￼ +    sed -i "s/DOM0_PIN_CPUS/2-5/g" ${D}${sysconfdir}/init.d/dom0_vcpu_pin.sh
}

do_install_append_salvator-x-h3-xt() {
￼ +    # As M3 has only 2 A57 cores we have to modify files where we pin VCPUs for certain domains
￼ +
￼ +    # Fixup a number of PCPUs the VCPUs of Dom0 must run on
￼ +    sed -i "s/DOM0_PIN_CPUS/4-7/g" ${D}${sysconfdir}/init.d/dom0_vcpu_pin.sh
}
```

**Comment by andr2000** [comment]
`Reviewed-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>`

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/dom0_vcpu_pin.sh [hunk 1]
```diff
@@ -7,6 +7,6 @@
 # Default-Stop:
 ### END INIT INFO
 
-# Force all VCPUs of Domain-0 to only run on PCPUs from 4 to 7 (A53 cores)
+# Force all VCPUs of Domain-0 to only run on DOM0_ALLOWED_PCPUS PCPUs (A53 cores)
```
Comment (lorc) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/dom0_vcpu_pin.sh:10]: After replacement comment will say `# Force all VCPUs of Domain-0 to only run on 4-7s (A53 cores)`

Comment (lorc) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/dom0_vcpu_pin.sh:10]: Ah, sorry. Ignore this.

**Comment by lorc** [comment]
`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Pull request merged**